### ISSUE TITLE
Feature request list

### ISSUE THREAD

**Issue by LiquidFun** [LiquidFun/aoc_tiles#1]
* Features
  * [X] Flag: auto-add-tiles-to-git
  * [ ] Flag: auto-add .aoc-tiles to .gitignore
  * [ ] Flag: auto-add `<!-- AOC BEGIN...` to README and ensure it exists
  * [x] Flag: highlight top 100 placement
  * [x] Total star count across all years

* Performance
  * [X] Multiprocessing
  * [X] More efficient yaml loader (CLoader instead of FullLoader)
  * [X] More efficient git-ignore check
  * [ ] Cache if tiles have already been created

* Documentation
  * [x] Add installation instructions to readme for pre-commit hook
  * [ ] List configuration options with images in readme
  * [ ] Improve output to user (what was added to git, how many tiles were regenerated)

* Tests
  * [ ] Automatically test aoctiles for some repositories

* Fix
  * [ ] Debug logging at the beginning even if verbose not set
  * [x] Git add/amend README as well
 
* CICD
  * [ ] Single-step update version in pyproject, create release, and push to pypi with github actions

**Comment by tonyaellie** [comment]
This project is really cool, thanks for creating it, a total star count would be nice
also are these not the wrong way round in the readme?
![image](https://github.com/LiquidFun/aoc_tiles/assets/46281725/a59d374b-92d0-4990-adfa-a2244e561f90)

**Comment by LiquidFun** [comment]
Good idea with the total star count! I'll try adding it, although not in the next days.

Thanks, I've fixed the swapped images!

**Comment by LiquidFun** [comment]
@tonyaellie I've added a total star count header in 0.6.0

**Comment by tonyaellie** [comment]
yay, thank you 💜

**Comment by abnew123** [comment]
Hey, big fan of this and I've integrated it with my 2023 repo (https://github.com/abnew123/aoc2023). However, I'm trying to do some prework to set up for my 2024 repo and it's slightly annoying that every single commit I'm getting a "AssertionError: Found no leaderboard?!" error. I get that it's true there's no leaderboard yet, maybe there's a way to add a suppression flag in config which will instead turn aoc_tiles into something that says "Coming soon"? Or ideally something dynamic in leaderboard.py which will check it's a valid leaderboard outside of the year number and do it automatically.

**Comment by LiquidFun** [comment]
hey @abnew123, sorry for taking so long, not sure if you need it anymore. But I've fixed it in version 0.6.3

**Comment by abnew123** [comment]
> hey @abnew123, sorry for taking so long, not sure if you need it anymore. But I've fixed it in version 0.6.3

Thanks! unfortunately I think that wasn't my actual issue in https://github.com/abnew123/aoc2024 (it's still saying "Found no leaderboard?!" even though the 2024 leaderboard is now open), so it was probably just user error haha.

**Comment by LiquidFun** [comment]
Hmm do you still have the issue? 

Have you updated to 0.6.3 in the pre-commit config? (the version on github has 0.6.2)
And you could try deleting .aoc-tiles/cache, maybe some invalid leaderboard was downloaded.

**Comment by abnew123** [comment]
Sorry yeah I hadn't pushed yet but did try updating (I've pushed now). And yes I've tried deleting the cache before, it comes back with the https://adventofcode.com/2024/leaderboard page html rather than the https://adventofcode.com/2024/leaderboard/self page it seems.

**Comment by LiquidFun** [comment]
Your repo works for me. Can you update you session.cookie, maybe it is too old? They usually last only 30 days. If that's the case I really have to update the error message in that case

**Comment by abnew123** [comment]
That worked thanks! One note is that it was quite painful to actually get the tiles up even after the cookie was good though, I ended up needing to delete the cache, the tiles folder and refresh my README file before the changes would finally take. Might have just been weird stored state in my IDE/git though.

**Comment by LiquidFun** [comment]
That's weird. I think deleting the cache should have been enough. Not sure why. But I'll look into, and I'll see if I can make the error better.

**Comment by tonyaellie** [comment]
Probably out of scope for the project but support for [everybody codes](https://everybody.codes/)?

**Comment by LiquidFun** [comment]
> Probably out of scope for the project but support for [everybody codes](https://everybody.codes/)?

I'll see if it has an API which can be adapted to this tool. But no promises on a timeline

**Comment by Rietty** [comment]
Hi! Big fan of your project and absolutely love it. Can we get an option to override colours for specific languages? For example I love the design/stripes but would like to use a colour that isn't blue for Python, perhaps a pink or a green for example.

Another thing that would be useful would be the ability to specify a different/custom folder. (This is just me being weird) But I _really_ _really_ hate how `.aoc_tiles` looks in a directory... if we could rename that to something else like `media` or something it would be nice.

**Comment by tonyaellie** [comment]
Use https://github.com/topaz/aoc-tmp-stats to guestimate position based on time (for 2025)

**Comment by LiquidFun** [comment]
@tonyaellie I like that idea. I wonder if it's against the spirit of the new rules though:

`(However, I've made it so you can share a read-only view of your private leaderboard. Please don't use this feature or data to create a "new" global leaderboard.)`

Although the data is provided by topaz himself, so maybe it's ok.

**Comment by tonyaellie** [comment]
@LiquidFun I think the fact that Topaz is providing the data and it's not really a leaderboard more just allowing you to see your personal position post competition means it really doesn't violate the request.
